web-app/react-app/components/ProfilePage/IntegrationKeyHandler/IntegrationKeyHandlerSegment/index.jsx

```diff
@@ -74,7 +78,7 @@ export class IntegrationKeyHandlerSegment extends Component<Props> {
                     />
                     <IntegrationKeyHandlerInput
                         fields={this.props.inputFields}
-                        onNew={this.onNew}
+                        onNew={this.props.onNew || this.onNew}
```

Comment:
Getting the `onNew` function as a prop is the right way to go👍🏻 (we/I need to change it also in other parts of the code)